tests/apps/search/test_indexers_courses.py

```diff
@@ -983,3 +985,103 @@ def test_indexers_courses_format_es_document_for_autocomplete(self):
+            catalog_visibility=CourseRunCatalogVisibility.HIDDEN,
+        )
+        self.assertTrue(course.extended_object.publish("en"))
+        indexed_courses = list(
+            CoursesIndexer.get_es_documents(index="some_index", action="some_action")
+        )
+        self.assertEqual(len(indexed_courses), 1)
+        self.assertEqual(indexed_courses[0]["course_runs"], [])
+
+    def test_indexers_courses_get_es_documents_course_run_catalog_visibility_about(
+        self,
+    ):
+        """
+        A course run with `about` on catalog visibility should not be indexed.
+        """
+        course = CourseFactory()
+        CourseRunFactory(
+            direct_course=course,
+            catalog_visibility=CourseRunCatalogVisibility.COURSE_ONLY,
+        )
+        self.assertTrue(course.extended_object.publish("en"))
+        indexed_courses = list(
+            CoursesIndexer.get_es_documents(index="some_index", action="some_action")
+        )
+        self.assertEqual(len(indexed_courses), 1)
+        self.assertEqual(indexed_courses[0]["course_runs"], [])
+
+    def test_indexers_courses_get_es_documents_course_run_catalog_visibility_both(self):
+        """
+        A course run with `both` on catalog visibility should be indexed.
+        """
+        course = CourseFactory()
+        CourseRunFactory(
+            direct_course=course,
+            catalog_visibility=CourseRunCatalogVisibility.COURSE_AND_SEARCH,
+        )
+        self.assertTrue(course.extended_object.publish("en"))
+        indexed_courses = list(
+            CoursesIndexer.get_es_documents(index="some_index", action="some_action")
+        )
+        self.assertEqual(len(indexed_courses), 1)
+        self.assertEqual(len(indexed_courses[0]["course_runs"]), 1)
+
+    def test_indexers_courses_get_es_documents_course_catalog_visibility_hidden_and_both(
+        self,
+    ):
+        """
+        A course with one run with `hidden` and another with `both` on catalog visibility should
+        only have a single run on the index.
+        """
+        course = CourseFactory()
+        CourseRunFactory(
+            direct_course=course,
+            catalog_visibility=CourseRunCatalogVisibility.HIDDEN,
+        )
+        CourseRunFactory(
+            direct_course=course,
+            catalog_visibility=CourseRunCatalogVisibility.COURSE_AND_SEARCH,
+        )
+        self.assertTrue(course.extended_object.publish("en"))
+        indexed_courses = list(
+            CoursesIndexer.get_es_documents(index="some_index", action="some_action")
+        )
+        self.assertEqual(len(indexed_courses), 1)
+        self.assertEqual(len(indexed_courses[0]["course_runs"]), 1)
+
+    def test_indexers_courses_get_es_documents_course_catalog_visibility_one_each(self):
+        """
+        A course with 3 runs. A run with `hidden`, another with `both` and a last one with `about`
+        on the catalog visibility should have a single run on the index.
```

Comment:
you can now update the visibility values

Reply:
Yes, I forgot to change the comments. I've also renamed some test method names because they included the `both` or `about` string.